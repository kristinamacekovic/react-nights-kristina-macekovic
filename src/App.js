import React, { Component } from 'react'
import { Switch, Route } from 'react-router-dom'
import { Provider } from 'react-redux'

import GlobalStyles from './GlobalStyles'
import { H1 } from './components/Typography'
import { ProductList } from './pages/ProductList'
import { ProductDetail } from './pages/ProductDetail'
import { Cart } from './pages/Cart'
import { SignUp } from './pages/SignUp'
import store from './store'

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <GlobalStyles />
        <H1>Welcome to the Purple Brand Store</H1>
        <Switch>
          <Route path="/" exact component={ProductList} />
          <Route path="/cart" component={Cart} />
          <Route path="/signup" component={SignUp} />
          <Route path="/:productID" component={ProductDetail} />
        </Switch>
      </Provider>
    )
  }
}

export default App
