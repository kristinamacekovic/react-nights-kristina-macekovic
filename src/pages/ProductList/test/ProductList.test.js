import React from 'react'
import { waitForElement } from 'react-testing-library'
import 'jest-styled-components'

import { App } from '../../../App'
import * as routes from '../../../routes'
import { renderWithRouter } from '../../../utilsTests/render'
import { mockFetchProducts } from '../../../utilsTests/mockHelpers'
import { PRODUCTS } from '../../../utilsTests/mockData'

describe('[pages] ProductsList', () => {
  describe('when loading', () => {
    it('should render correctly', () => {
      const renderer = renderWithRouter(<App />, routes.PRODUCT_LIST)
      expect(renderer.container).toMatchSnapshot()
    })
  })

  describe('when products loaded', () => {
    mockFetchProducts()

    it('should render correctly', async () => {
      const renderer = renderWithRouter(<App />, routes.PRODUCT_LIST)
      await waitForElement(() => renderer.getByTestId('product-in-list'))
      expect(renderer.getAllByTestId('product-in-list').length).toEqual(
        PRODUCTS.data.length
      )
    })
  })
})