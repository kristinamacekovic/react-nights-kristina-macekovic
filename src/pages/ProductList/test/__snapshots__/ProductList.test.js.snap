// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] ProductsList when loading should render correctly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <h2>
    Sorry, there is an error, please try refreshing the page.
  </h2>
</div>
`;
